File: view/src/lib/types/DamageType.ts
export enum DamageType {
  Acid = "acid",
  Bludgeoning = "bludgeoning",
  Cold = "cold",
  Fire = "fire",
  Force = "force",
  Lightning = "lightning",
  Necrotic = "necrotic",
  NonMagicalBludgeoning = "non_magical_bludgeoning",
  NonMagicalPiercing = "non_magical_piercing",
  NonMagicalSlashing = "non_magical_slashing",
  Piercing = "piercing",
  Poison = "poison",
  Psychic = "psychic",
  Radiant = "radiant",
  Slashing = "slashing",
  Thunder = "thunder",
}

export const DisplayDamageType: Readonly<Record<DamageType, string>> = {
  [DamageType.Acid]: "Acid",
  [DamageType.Bludgeoning]: "Bludgeoning",
  [DamageType.Cold]: "Cold",
  [DamageType.Fire]: "Fire",
  [DamageType.Force]: "Force",
  [DamageType.Lightning]: "Lightning",
  [DamageType.Necrotic]: "Necrotic",
  [DamageType.NonMagicalBludgeoning]: "Non-magical Bludgeoning",
  [DamageType.NonMagicalPiercing]: "Non-magical Piercing",
  [DamageType.NonMagicalSlashing]: "Non-magical Slashing",
  [DamageType.Piercing]: "Piercing",
  [DamageType.Poison]: "Poison",
  [DamageType.Psychic]: "Psychic",
  [DamageType.Radiant]: "Radiant",
  [DamageType.Slashing]: "Slashing",
  [DamageType.Thunder]: "Thunder",
} as const;

export const DamageTypes: DamageType[] = Object.values(DamageType);

File: api/src/types/damage_type.rs
#[derive(Debug, Clone, Copy, PartialEq, sqlx::Type, serde::Serialize, serde::Deserialize)]
#[sqlx(type_name = "damage_type", rename_all = "snake_case")]
#[serde(rename_all = "snake_case")]
pub enum DamageType {
    Acid,
    Bludgeoning,
    Cold,
    Fire,
    Force,
    Lightning,
    Necrotic,
    NonMagicalBludgeoning,
    NonMagicalPiercing,
    NonMagicalSlashing,
    Piercing,
    Poison,
    Psychic,
    Radiant,
    Slashing,
    Thunder,
}
